src/Aspire.Hosting.AWS/CloudFormation/CloudFormationExtensions.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Amazon.CloudFormation;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS;
+using Aspire.Hosting.AWS.CloudFormation;
+using Aspire.Hosting.Lifecycle;
+
+namespace Aspire.Hosting;
+
+public static class CloudFormationExtensions
+{
+    /// <summary>
+    /// Add a CloudFormation stack for provisioning application resources.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="stackName">The name of the CloudFormation stack.</param>
+    /// <param name="templatePath">The path to the CloudFormation template that defines the CloudFormation stack.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ICloudFormationResource> AddAWSCloudFormationTemplate(this IDistributedApplicationBuilder builder, string stackName, string templatePath)
+    {
+        var resource = new CloudFormationResource(stackName, templatePath);
+        var cfBuilder = builder.AddResource(resource);
+
+        builder.Services.TryAddLifecycleHook<CloudFormationLifecycleHook>();
+        return cfBuilder;
+    }
+
+    /// <summary>
+    /// The AWS SDK service client configuration used to create the CloudFormation service client.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="awsSdkConfig">The name of the AWS credential profile.</param>
+    public static IResourceBuilder<ICloudFormationResource> WithReference(this IResourceBuilder<ICloudFormationResource> builder, IAWSSDKConfig awsSdkConfig)
+    {
+        builder.Resource.AWSSDKConfig = awsSdkConfig;
+        return builder;
+    }
+
+    /// <summary>
+    /// Override the CloudFormation service client the ICloudFormationResource would create to interact with the CloudFormation service. This can be used for pointing the
+    /// CloudFormation service client to a non-standard CloudFormation endpoint like an emulator.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="cloudFormationClient">The AWS CloudFormation service client.</param>
+    public static IResourceBuilder<ICloudFormationResource> WithReference(this IResourceBuilder<ICloudFormationResource> builder, IAmazonCloudFormation cloudFormationClient)
+    {
+        builder.Resource.CloudFormationClient = cloudFormationClient;
+        return builder;
+    }
+
```

Comment:
This pattern is unusual. WithReference applies to another resource. Maybe these can be specific calls to `WithResourceConfiguration` or something similar.

Reply:
I agree using `WithReference` isn't a great fit. I'm basically trying to handle adding optional configuration information to the `ICloudFormationResource`. I also want to leave myself some room in case I want to add more optional configurations in the future. I prefer using a fluent style versus an "options" type at the end of the `AddAWSCloudFormationTemplate`.